### PR TITLE
Host and realm use the same logic for authenticating against a realm

### PR DESCRIPTION
**Pull request by jurgenwerk**
This PR consolidates the logic where we use matrix rooms to perform authentication between the 2 parties which results in a JWT.

Before this PR, the logic was performed in two different places but with different code which was essentially doing the same thing.

### CODE REVIEW

File: packages/runtime-common/matrix-client.ts [hunk 1]
```diff
@@ -13,10 +13,14 @@ export class MatrixClient {
     private password: string,
   ) {}
 
-  get userId() {
+  getUserId() {
```
Comment (jurgenwerk) [inline packages/runtime-common/matrix-client.ts:16]: This was changed to be compatible with the `RealmAuthMatrixClientInterface`

File: packages/realm-server/tests/index.ts [hunk 2]
```diff
@@ -4,3 +4,4 @@ import './loader-test';
 import './indexing-test';
 import './module-syntax-test';
 import './permissions/permission-checker-test';
+import './auth-client-test';
```
Comment (jurgenwerk) [inline packages/realm-server/tests/index.ts:7]: The auth client test wasn't even included in the test suite 😱

Comment (backspace) [reply packages/realm-server/tests/index.ts:7]: wow, great catch, I think we should add linting to prevent this happening 😳

**Review by backspace — APPROVED** [review]
nice improvement 🤩

**Review by habdelra — APPROVED** [review]

**Comment by jurgenwerk** [comment]
This PR has test failures but that is from upstream and unrelated to this. Merging it anyway and we will address that in another PR.

**Pull request merged**